Pod/Classes/Forms/FieldValidator.swift

```diff
@@ -0,0 +1,115 @@
+//
+//  FieldValidator.swift
+//  Swiftilities
+//
+//  Created by Nicholas Bonatsakis on 5/25/16.
+//  Copyright © 2016 Raizlabs. All rights reserved.
+//
+
+import Foundation
+
+struct FieldValidator {
+    let fieldName: String
+    let rules: [FieldValidationRule]
+    
+    func validate(inputString: String) -> ErrorType? {
+        for rule in rules {
+            if let error = rule.validate(fieldName, inputString: inputString) {
+                return error
+            }
+        }
+        
+        return nil
+    }
+}
+
+protocol FieldValidationRule {
+    func validate(fieldName: String, inputString: String) -> ErrorType?
+}
+
+enum ValidationRule: FieldValidationRule {
+    case NonEmpty
+    case MinLength(length: Int)
+    case ValidEmail
+    
```

Comment:
#nth Add a `Custom` case that would take in a block for an argument and use that block as the validation. 

This would allow an object to just hold a variable of type `ValidationRule` even if it is some type of validator that we haven't come up with. 

The only major complication that I can see arising from this is that you may need to create a new custom errortype for each `Custom` Validation rule is custom, but that shouldn't be THAT big of a deal.